server/gitrest/packages/gitrest-base/src/utils/helpers.ts

```diff
@@ -119,18 +120,30 @@ export async function retrieveLatestFullSummaryFromStorage(
 /**
  * Retrieves the full repository path. Or throws an error if not valid.
  */
-export function getRepoPath(name: string, owner?: string): string {
-    // `name` needs to be always present and valid.
-    if (!name || path.parse(name).dir !== "") {
-        throw new NetworkError(400, `Invalid repo name provided.`);
+export function getRepoPath(
+    repoPerDocEnabled: boolean,
```

Comment:
Can we get away with `getRepoPath` not having to know about `repoPerDoc`? Perhaps if `documentId` were simply optional, and is used if not undefined? I think it could help simplify this function a lot.